File: leetcode/python/371.py
## Leetcode 371 - Sum of Two Integers
## Nov. 29, 2023

class Solution:
    def getSum(self, a: int, b: int) -> int:
        mask = 0xffffffff
        while (b & mask) > 0:
            c = a & b
            a ^= b
            b = c << 1
        return a & mask if b > mask else a

File: leetcode/c/371.c
// Leetcode 371 - Sum of Two Integers
// Nov. 29, 2023

int getSum(int a, int b) {
    unsigned c;
    while (b) {
        c = a & b;
        a ^= b;
        b = c << 1;
    }
    return a;
}